File: server-java/src/main/java/com/conarum/tradingjournal/domain/journal/model/JournalEntry.java
package com.conarum.tradingjournal.domain.journal.model;

import com.conarum.tradingjournal.common.model.BaseEntity;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@Document(collection = "journalentries")
@CompoundIndex(name = "date_userEmail_unique", def = "{'date': 1, 'userEmail': 1}", unique = true)
public class JournalEntry extends BaseEntity {
    private String date; // Format: YYYY-MM-DD
    private String content = "";
    private String mood = "neutral";
    private String preMarketNotes = "";
    private String postMarketNotes = "";
    private String marketCondition = "";
    private boolean isChecklistDone = false;
}

File: server/src/models/JournalEntry.ts
import mongoose, { Schema } from 'mongoose';
import { v4 as uuidv4 } from 'uuid';

export interface IJournalEntry {
  _id: string;
  date: string;
  content: string;
  mood: string;
  preMarketNotes: string;
  postMarketNotes: string;
  marketCondition: string;
  isChecklistDone: boolean;
  userEmail: string;
  createdAt: string;
  updatedAt: string;
}

const JournalEntrySchema = new Schema<IJournalEntry>({
  _id: { type: String, default: () => uuidv4() },
  date: { type: String, required: true },
  content: { type: String, default: '' },
  mood: { type: String, default: 'neutral' },
  preMarketNotes: { type: String, default: '' },
  postMarketNotes: { type: String, default: '' },
  marketCondition: { type: String, default: '' },
  isChecklistDone: { type: Boolean, default: false },
  userEmail: { type: String, required: true },
  createdAt: { type: String, default: () => new Date().toISOString() },
  updatedAt: { type: String, default: () => new Date().toISOString() },
});

JournalEntrySchema.index({ date: 1, userEmail: 1 }, { unique: true });

export const JournalEntry = mongoose.model<IJournalEntry>('JournalEntry', JournalEntrySchema);
